docs/en/stores/basic-usage.md

```diff
@@ -0,0 +1,454 @@
+const fetchUser = createCommand(async ({ at, get, payload: { username, password } }) => {
+	const token = await fetchToken(username, password);
+
+	return [
+		add(path('auth', 'token'), token);
+	];
+}
+
+const loadUserData = createCommand(async ({ path }) => {
+	const token = get(path('auth', 'token'));
+	const user = await fetchCurrentUser(token);
+	return [
+		replace(path('users', 'current'), user)
+	];
+});
+
+export const login = createProcess('login', [ fetchUser, loadUserData ]);
+```
+
+#### Middleware
+
+Middleware wraps a process in `before` and `after` callbacks that surround a process. A common use of middleware is for the handling of errors and reinstating consistent state.
+
+Lets add some middleware that runs after the commands and resets the state if an error occurs.
+
+> commands/login.ts example
+
+```ts
+import { createCommandFactory } from "@dojo/framework/stores/process";
+import { State } from './interfaces';
+import { add, replace } from "@dojo/framework/stores/state/operations";
+
+const createCommand = createCommandFactory<State>();
+
+const fetchUser = createCommand(async ({ at, get, payload: { username, password } }) => {
+	const token = await fetchToken(username, password);
+
+	return [
+		add(path('auth', 'token'), token);
+	];
+}
+
+const loadUserData = createCommand(async ({ path }) => {
+	const token = get(path('auth', 'token'));
+	const user = await fetchCurrentUser(token);
+	return [
+		replace(path('users', 'current'), user)
+	];
+});
+
+const resetUserOnError = () => {
+	return {
+		after: (error, result) => {
+			if (error) {
+				result.store.apply([
+					remove(path('auth', 'token')),
+					remove(path('users', 'current'))
+				])
+			}
+		}
+	};
+};
+
+const login = createProcess('login', [ fetchUser, loadUserData ], [ resetUserOnError ]);
+```
+
+#### `payload` type
+
+The process executor's `payload` is inferred from the `payload` type of the commands. If the payloads differ then it is necessary to explictly define the `payload` type.
+
+> Explicitly defining `payload` type
+
+```ts
+const createCommand = createCommandFactory<State>();
+
+const commandOne = createCommand<{ one: string }>(({ payload }) => {
+	return [];
+});
+
+const commandTwo = createCommand<{ two: string }>(({ payload }) => {
+	return [];
+});
+
+const process = createProcess<State, { one: string; two: string }>('example', [commandOne, commandTwo]);
+
+process(store)({ one: 'one', two: 'two' });
+```
+
+## Widgets and Stores
+
+There are two state containers available for widgets: `Container` and `StoreProvider`. These containers connect the application store with a widget.
+
+Note that the documentation in this section is intended to show how widgets and state are connected. For a more detailed look see the [Widgets]() reference guide.
+
+### StoreProvider
+
+A `StoreProvider` is a widget that has its own `renderer` and connects to the store. It is always encapsulated in another widget because a it does not have properties defined.
```

Comment:
```suggestion
A `StoreProvider` is a Dojo widget that has its own `renderer` and connects to the store. It is always encapsulated in another widget because it does not defined its own properties.
```